webapp/security/views.py

```diff
@@ -352,171 +352,148 @@ def cve(cve_id):
-                )
-            )
 
-    if data.get("references"):
-        for ref in data["references"]:
-            references.append(CVEReference(uri=ref))
-
-    if data.get("bugs"):
-        for b in data["bugs"]:
-            bugs.append(Bug(uri=b))
-
-    cve = CVE(
-        id=data["id"],
-        status=data.get("status", ""),
-        last_updated_date=parse(data.get("last_updated_date"))
-        if data.get("last_updated_date")
-        else None,
-        public_date_usn=parse(data.get("last_updated_date"))
-        if data.get("public_date_usn")
-        else None,
-        public_date=parse(data.get("public_date"))
-        if data.get("public_date")
-        else None,
-        priority=data.get("priority"),
-        cvss=data.get("cvss"),
-        assigned_to=data.get("assigned_to"),
-        discovered_by=data.get("discovered_by"),
-        approved_by=data.get("approved_by"),
-        description=data.get("description"),
-        ubuntu_description=data.get("ubuntu_description"),
-        notes=data.get("notes", []),
-        packages=packages,
-        references=references,
-        bugs=bugs,
-    )
+        if data.get("references"):
+            for ref in data["references"]:
+                references.append(CVEReference(uri=ref))
+
+        if data.get("bugs"):
+            for b in data["bugs"]:
+                bugs.append(Bug(uri=b))
+
+        cve = CVE(
+            id=data["id"],
+            status=data.get("status", ""),
+            last_updated_date=parse(data.get("last_updated_date"))
+            if data.get("last_updated_date")
+            else None,
+            public_date_usn=parse(data.get("public_date_usn"))
+            if data.get("public_date_usn")
+            else None,
+            public_date=parse(data.get("public_date"))
+            if data.get("public_date")
+            else None,
+            priority=data.get("priority"),
+            cvss=data.get("cvss"),
+            assigned_to=data.get("assigned_to"),
+            discovered_by=data.get("discovered_by"),
+            approved_by=data.get("approved_by"),
+            description=data.get("description"),
+            ubuntu_description=data.get("ubuntu_description"),
+            notes=data.get("notes", []),
+            packages=packages,
+            references=references,
+            bugs=bugs,
+        )
+        cve_object.append(cve)
+    return cve_object
 
-    try:
-        db_session.add(cve)
-        db_session.commit()
-    # Duplicate key errors and Payload input errors
-    except (IntegrityError, DataError) as error:
-        return flask.jsonify({"message": error.orig.args[0]}), 409
 
-    return flask.jsonify({"message": "CVE created succesfully"}), 201
+def update_cve(cve_data):
 
+    cve_object = []
 
-@authorization_required
-def update_cve(cve_id):
-    data = flask.request.json
-    cve = db_session.query(CVE).get(cve_id)
+    for data in cve_data:
+        cve = db_session.query(CVE).get(data["id"])
```

Comment:
You're doing this query twice - you already did it on line 539. You should just pass through that `cve` object rather than getting it again.